### PR TITLE
Start notes for Sod's shock tube problem.

### PR DESCRIPTION
**Pull request by yungyuc**
@yuhow This is a very rough starting point of a Sod's shock tube note.  The whole chapter 7 in [1] is discussing the problem.  I haven't decided how I'd like to treat it, and suspended my work last autumn.  But some of the diagrams I did might be helpful to you.

You can pull the branch and check the results.  If you see things are fine please go ahead merge this request.  Thanks.

[1] J. D. Anderson, Modern Compressible Flow: With Historical Perspective, 3rd ed. Boston: McGraw-Hill, 2003. http://www.amazon.com/Modern-Compressible-Flow-Historical-Perspective/dp/0072424435

### CODE REVIEW

**Comment by yuhow** [comment]
OK, I will take a look. Thanks!

**Comment by yuhow** [comment]
Well done!

**Comment by yuhow** [comment]
One quick question: why are the subscripts of gamma in your figure 1 and figure 2 not consistent to each other? I though the value gamma is same in both region.

File: shocktube/schematic/tube_move_right.tex [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+\psline[linewidth=0.5pt,linestyle=dashed]{}(0,-0.7)(0,0.7)
+\psline[linewidth=0.5pt,linestyle=solid,arrowsize=4pt 2]{->}(0, 0.4)(0.4, 0.4)
+\psline[linewidth=0.5pt,linestyle=solid,arrowsize=4pt 2]{->}(0, 0  )(0.4, 0  )
+\psline[linewidth=0.5pt,linestyle=solid,arrowsize=4pt 2]{->}(0,-0.4)(0.4,-0.4)
+\rput[l](0.5,0){$u_f$}
+
+% Flow properties.
+\rput[l](-4.9,0.2){$p_4, \rho_4, \gamma_4$}
+\rput[l](-4.9,-0.2){$u_4 = 0$}
+\rput[r](4.9,0.2){$p_1, \rho_1, \gamma_4$}
```
Comment (yungyuc) [inline shocktube/schematic/tube_move_right.tex:21]: $\gamma_4$ should be $\gamma_1$.  It's a typo.

**Comment by yungyuc** [comment]
Answered inline

**Comment by yuhow** [comment]
Second question: I feel quite confused about the definition of 'driving gas' and 'driven gas'. When I wrote my note, I referred to this book: http://www.astro.uu.se/~hoefner/astro/teach/ch10.pdf
And it mentioned that:
_The region with the highest pressure is called the driven section of the tube, while the low-pressure part is the working section_

It seems we maybe also have to collaborate on defining objects.

**Comment by yungyuc** [comment]
Anderson defined otherwise: the high pressure is the driver section and the low pressure is the driven section.  As long as we make it clear it's fine.  The important thing is to keep the notes clear.

**Comment by yuhow** [comment]
No problem.

**Comment by yungyuc** [comment]
Prof. Höfner's note (you posted) is excellent.  It contain classical numerical methods for the shock tube problem.  Inspecting the diagram you'll see the result you obtained from the CESE method is much better than those diagrams.  More importantly, the CESE method doesn't use an exact solution of the Riemann problem nor the eigenvalues to capture the shock, and that's the attractive point of the CESE method.

**Pull request merged**